ast/expressions.go

```diff
@@ -228,11 +229,29 @@ func (n *CaseExpr) Accept(v Visitor) (Node, bool) {
 	return v.Leave(n)
 }
 
+// SubQuery represents a subquery interface.
+// This interface is implemented in executor and used in plan/evaluator.
+type SubQuery interface {
+	ExprNode
+
+	// EvalRows executes the subquery and returns the multi rows with rowCount.
+	// rowCount < 0 means no limit.
+	// If the ColumnCount is 1, we will return a column result like {1, 2, 3},
+	// otherwise, we will return a table result like {{1, 1}, {2, 2}}.
+	EvalRows(ctx context.Context, rowCount int) ([]interface{}, error)
+
+	// ColumnCount returns column count for the sub query.
+	ColumnCount() (int, error)
+}
+
 // SubqueryExpr represents a subquery.
 type SubqueryExpr struct {
 	exprNode
 	// Query is the query SelectNode.
-	Query ResultSetNode
+	Query           ResultSetNode
+	SubQuery        SubQuery
```

Comment:
Seems we should use a new name, SubQuery and SubqueryExpr are a little confused.